src/coreclr/src/jit/emitxarch.cpp

```diff
@@ -12593,9 +12632,239 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
+                        unsigned nMaxPaddingBytes = (1 << (maxBlocksAllowedForLoop - minBlocksNeededForLoop + 1)) - 1;
+                        unsigned nPaddingBytes    = (-(int)(size_t)dst) & (alignmentBoundary - 1);
+
+                        // Check if the loop exceed maxSize
+                        if (loopSize > maxLoopSize)
+                        {
+                            skipPadding = true;
+                            assert(!"Should never hit maxLoopSize threshold because it should have been predicted "
+                                    "earlier.");
+                        }
+
+                        // Check if the alignment exceeds maxPadding limit
+                        else if (nPaddingBytes > nMaxPaddingBytes)
+                        {
+                            // Cannot align to 32B, so try to align to 16B boundary.
+                            alignmentBoundary >>= 1;
+                            nMaxPaddingBytes  = 1 << (maxBlocksAllowedForLoop - minBlocksNeededForLoop + 1);
+                            nPaddingBytes     = (-(int)(size_t)dst) & (alignmentBoundary - 1);
+
+                            // Check if the loop is already at new alignment boundary
+                            if (nPaddingBytes == 0)
+                            {
+                                skipPadding = true;
+#if DEBUG
+                                if (displayAlignmentDetails)
+                                {
+                                    printf("\t\t;; Skip alignment: 'Loop already aligned at 16B boundary.' in (%s)\n",
+                                           emitComp->info.compMethodName);
+                                }
+#endif
+                            }
+                            // Check if the alignment exceeds new maxPadding limit
+                            else if (nPaddingBytes > nMaxPaddingBytes)
+                            {
+                                skipPadding = true;
+#if DEBUG
+                                if (displayAlignmentDetails)
+                                {
+                                    printf("\t\t;; Skip alignment: 'PaddingNeeded= %d, MaxPadding= %d, LoopSize= %d, "
+                                           "AlignmentBoundary= %dB.' in (%s)\n",
+                                           nPaddingBytes, nMaxPaddingBytes, loopSize, alignmentBoundary,
+                                           emitComp->info.compFullName);
+                                }
+#endif
+                            }
+                        }
+
+                        if (!skipPadding)
+                        {
+                            // Padding is needed only if loop starts at or after the current offset.
+                            // Otherwise, the loop just fits in minBlocksNeededForLoop and so can skip alignment.
+                            size_t extraBytesNotInLoop = (size_t)(32 * minBlocksNeededForLoop) - loopSize; // For calculation, use 32B chunks
```

Comment:
Seems a bit odd to just see `32` here like this when everything else is symbolic?

Reply:
Sure. Replaced it with `emitComp->opts.compJitAlignLoopBoundary`.